jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -223,6 +228,19 @@ def _on_add_data_to_viewer(self, msg):
         if isinstance(msg.data, glue_core_data):
             data_obj = None
 
+        # sync viewer and UC plugin units when new data is loaded. this is necessary, for
+        # example, when all data is unloaded but unit selections are set in the UC plugin,
+        # and new data is loaded which should use those units
+        if len(self.spectral_unit_selected):
+            if hasattr(viewer.state, 'x_display_unit') and len(self.spectral_unit_selected):
+                if viewer.state.x_display_unit != self.spectral_unit_selected:
+                    xunit = _valid_glue_display_unit(self.spectral_unit.selected, viewer, 'x')
+                    viewer.state.x_display_unit = xunit
+                    viewer.set_plot_axes()
```

Comment:
this probably would require a line-break for codestyle anyways... so feel free to keep 2 if statements if you prefer, but I think we can at least get rid of the second check for `len(self.spectral_unit_selected)`

```suggestion
        if len(self.spectral_unit_selected) and hasattr(viewer.state, 'x_display_unit') and viewer.state.x_display_unit != self.spectral_unit_selected:
            xunit = _valid_glue_display_unit(self.spectral_unit.selected, viewer, 'x')
            viewer.state.x_display_unit = xunit
            viewer.set_plot_axes()
```